### PR TITLE
Added test project for the crypto module

### PR DESCRIPTION
**Pull request by haukepetersen**
The test_crypto project focuses so far only on testing the makefiles and module structure of the crypto module. It tests only if the crypto algorithms compile ok. For the future some explicit test vectors need to be added for each crypto function to validate their correct functionality.

### CODE REVIEW

File: test_crypto/main.c [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2008, 2009, 2010  Kaspar Schleiser <kaspar@schleiser.de>
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
```
Comment (mehlis) [inline test_crypto/main.c:14]: LGPL2 header missing

**Comment by mehlis** [comment]
@haukepetersen for the sha256 code I had a more advance functionality test: https://github.com/RIOT-OS/projects/blob/master/test_sha256/main.c#L35

would be awesome to have this for every module!

**Comment by haukepetersen** [comment]
I actually had something exactly like that in mind, just havn't found the time (and no linux/web crypto library that implements all our algorithmts) to produce the test vectors. It's something on the todo list, question if we keep this pull request open until I added test vectors to this test project or if I should make another pull request with the additions later on?!

**Comment by OlegHahm** [comment]
ACK and go.

**Pull request merged**